plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/connection/ConnectionsToStructRefactoring.java

```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Mathias Garstenauer - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.typemanagement.refactoring.connection;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.IdentifierVerifier;
+import org.eclipse.fordiac.ide.model.data.StructuredType;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetwork;
+import org.eclipse.fordiac.ide.model.libraryElement.FBType;
+import org.eclipse.fordiac.ide.model.libraryElement.ServiceInterfaceFBType;
+import org.eclipse.fordiac.ide.model.libraryElement.VarDeclaration;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibrary;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibraryManager;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.EditConnectionsChange;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.CompositeChange;
+import org.eclipse.ltk.core.refactoring.Refactoring;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+public class ConnectionsToStructRefactoring extends Refactoring {
+	private final URI sourceURI;
+	private final URI destinationURI;
+	private final URI netURI;
+	private final Map<String, String> replacableConMap;
+
+	private URI structURI;
+	private String sourceVarName;
+	private String destinationVarName;
+	private boolean conflictResolution;
+
+	private final Collection<VarDeclaration> vars;
+
+	private final TypeLibrary lib;
+
+	public ConnectionsToStructRefactoring(final FBNetwork net, final FBType sourceType, final FBType destinationType,
+			final Map<String, String> replacableConMap) {
+		this.netURI = EcoreUtil.getURI(net);
+		this.sourceURI = EcoreUtil.getURI(sourceType);
```

Comment:
what is the difference between source an destination ?

Reply:
I need to distinguish source and destination in order to know where to remove inputs or outputs and how to create the new struct connections